### PR TITLE
Adding XDG_CONFIG_HOME config file

### PR DESCRIPTION
**Pull request by thomasblanc**
As stated in the title of this PR, that should do it.

This solves #295.

### CODE REVIEW

File: src/indentConfig.ml [hunk 1]
```diff
@@ -406,11 +406,18 @@ let rec find_conf_file path =
...
-    try
-      let (/) = Filename.concat in
-      let f = (Sys.getenv "HOME") / ".ocp" / "ocp-indent.conf" in
-      if Sys.file_exists f then load ~indent:conf f else conf, [], []
-    with Not_found -> conf, [], []
+    let (/) = Filename.concat in
+    let getenv_f var path =
+      match Sys.getenv var with (* not using getenv_opt because should compile with old versions of OCaml *)
+      | x -> Some ( x / path )
+      | exception Not_found -> None (* not that old *)
```
Comment (dbuenzli) [inline src/indentConfig.ml]: None should also be returned if `x` is `""`.

**Comment by dbuenzli** [comment]
> This solves #295 though to be fully XDG-compliant you should print a warning when using the legacy config file.

Where does that come from ?

**Comment by thomasblanc** [comment]
> > This solves #295 though to be fully XDG-compliant you should print a warning when using the legacy config file.
> 
> Where does that come from ?

That comes from me being really bad at reading the XDG base directory specification and mixing up `XDG_RUNTIME_DIR` with `XDG_CONFIG_HOME`. Sorry about that.

File: src/indentConfig.ml (hunk 1 above)
Comment (thomasblanc) [reply src/indentConfig.ml]: Right, thanks!

**Comment by thomasblanc** [comment]
> This is not XDG compliant if `$XDG_CONFIG_HOME` is undefined or empty you need to fall back on `$HOME/.config` for the configuration directory.

Done in last commit.

File: src/indentConfig.ml [hunk 2]
```diff
@@ -406,11 +406,21 @@ let rec find_conf_file path =
...
-    with Not_found -> conf, [], []
+    let (/) = Filename.concat in
+    let getenv_f var path =
+      match Sys.getenv var with (* not using getenv_opt because should compile with old versions of OCaml *)
+      | "" -> None
+      | x -> Some ( x / path )
+      | exception Not_found -> None (* not that old *)
+    in
+    match getenv_f "XDG_CONFIG_HOME" ( "ocp" / "ocp-indent.conf" ), (* the XDG way *)
+          getenv_f "HOME" ( ".config" / "ocp" / "ocp-indent.conf" ), (* the XDG alternative *)
```
Comment (dbuenzli) [inline src/indentConfig.ml]: Somehow the XDG spec doesn't say anything about this (or says it in a very indirect manner) but I don't think this is the proper way of doing this. 

The reason is that you can't then disable a set of configuration files by setting your `XDG_CONFIG_HOME` to another directory. With the current code if you do, an exisiting `HOME/.config/ocp/ocp-indent` will be picked up which I think wouldn't be in the spirit of the XDG spec.

I think the right way of doing this is: 

1. Determine a configuration directory `$DIR` according to the XDG spec. If `$DIR/ocp/ocp-indent.conf` exists use this. 
2. If the previous step didn't give anything try (to maintain backward compat) `~/.ocp/ocp-indent.conf`.

Comment (thomasblanc) [reply src/indentConfig.ml]: My bad, fixing this.

**Comment by thomasblanc** [comment]
> I don't think that works. You are not checking the file exists. in XDG_CONFIG_HOME. Really implement the algorithm as I suggested it will be very clear what is going on.

I think this is what you asked for. It doesn't change the behavior (well, it no longer checks that `HOME` isn't `""` but that was the old behavior).

**Comment by dbuenzli** [comment]
> I think this is what you asked for.

I don't think so. I think in the previous iteration if you had XDG_CONFIG_HOME defined you'd never go in the legacy path. In any case even if it did actually worked the logic was encoded in a very obscure manner. It now looks clearer. Thanks !

**Comment by dbuenzli** [comment]
One last thing in the documentation only mention `$XDG_CONFIG_HOME` not `~/.config/` or `$HOME/.config`, it's implied by the XDG spec and makes the current documentation slightly wrong. 

Really the XDG spec for `XDG_CONFIG_HOME` is just: to find a configuration directory path use the value of `$XDG_CONFIG_HOME` and if it's unset or empty use `$HOME/.config` instead. In other words it doesn't define a file lookup procedure with fallbacks: it defines how to find a configuration directory by simply consulting environment variables.

**Comment by bcc32** [comment]
This PR broke the legacy path.  It used to be `$HOME/.ocp/ocp-indent.conf` but now it's `$HOME/.ocp/.ocp-indent.conf` (extra leading dot in filename).

**Pull request merged**